core/build/types.odin

```diff
@@ -5,31 +5,32 @@ import "core:runtime"
 Build_Command_Type :: enum {
 	Invalid,
 	Build,
+	Install,
 	Dev_Setup,
 	Display_Help,
 }
 
-Build_Options :: struct {
+Settings :: struct {
+	// Configurable via settings_init_from_args
 	command_type: Build_Command_Type,
 	config_name: string, // Config.name
 	dev_opts: Dev_Options,
-	display_external_configs: bool,
-	default_config_name: string,
-}
-
-
+	custom_args: [dynamic]string,
 
-Define_Val :: union #no_nil {
-	bool,
-	int,
-	string,
+	// Configurable in user code. 
+	display_external_configs: bool, // Allow the user to call configs from other build systems imported
+	default_config_name: string,    // Called when no config is specified in the args / config_name
 }
 
-Platform :: struct {
-	os: runtime.Odin_OS_Type,
-	arch: runtime.Odin_Arch_Type,
-}
 
+// Static lib?
```

Comment:
Called archive (as in, object archive), it just contains a collection of obj files. Though I doubt people would recognize it by that name. I don't think it's worth it: Odin isn't supposed to be built from multiple modules and then archived/linked together.

If you want to ship a static library from Odin, compile .obj and if it has external C dependencies, ship with those .lib dependencies (`foreign import`s should take care of linking). Also I don't think if you create an archive, that, `foreign import` would be able to find them.

I'm not at home so I can't check whether Odin supports creating archives. If not, then maybe it's not even worth it.

Reply:
![image](https://github.com/odin-lang/Odin/assets/31741257/e6f4036a-776d-4a35-ba93-884ccd36395f)
All I did was take the build modes from here (not the duplicate ones). It doesn't seem that it supports static libs.